src/Components/Health/Checker/HealthChecker/SwagSecurityChecker.php

```diff
@@ -0,0 +1,120 @@
+
+use Doctrine\DBAL\Connection;
+use Frosh\Tools\Components\Health\Checker\CheckerInterface;
+use Frosh\Tools\Components\Health\HealthCollection;
+use Frosh\Tools\Components\Health\SettingsResult;
+use Symfony\Bundle\FrameworkBundle\Console\Application;
+use Symfony\Component\Cache\Adapter\AdapterInterface;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Output\NullOutput;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+use Symfony\Component\HttpKernel\KernelInterface;
+
+class SwagSecurityChecker implements HealthCheckerInterface, CheckerInterface
+{
+    public function __construct(
+        private readonly Connection $connection,
+        private readonly KernelInterface $kernel,
+        #[Autowire('%kernel.shopware_version%')]
+        private readonly string $shopwareVersion,
+        #[Autowire(service: 'cache.object')]
+        private readonly AdapterInterface $objectCache
+    ) {}
+
+    public function collect(HealthCollection $collection): void
+    {
+        $this->refreshPlugins($this->connection);
+
+        try {
+            if ($this->hasSecurityAdvisories() === false) {
+                return;
+            }
+        } catch (\Throwable) {
+            $collection->add(
+                SettingsResult::error(
+                    'security-update',
+                    'Cannot check security.json from shopware-static-data',
+                    'not accessible',
+                    'accessible',
+                    'https://raw.githubusercontent.com/FriendsOfShopware/shopware-static-data/main/data/security.json'
+                )
+            );
+        }
+
+        if ($this->swagSecurityInstalled()) {
+            return;
+        }
+
+        $collection->add(
+            SettingsResult::error(
+                'security-update',
+                'Security update',
+                'Shopware outdated',
+                'Update Shopware to the latest version or install recent version of the plugin SwagPlatformSecurity',
+                'https://store.shopware.com/en/swag136939272659f/shopware-6-security-plugin.html'
+            )
+        );
+    }
+
+    private function refreshPlugins(Connection $connection): void
+    {
+        $result = $connection->executeQuery(
+            'SELECT COUNT(*) FROM plugin WHERE updated_at < DATE_SUB(NOW(), INTERVAL 1 DAY)'
+        )->fetchOne();
+
+        if ($result === 0 || $result === '0') {
+            return;
+        }
+
+        $pluginRefresh = new ArrayInput([
+            'command' => 'plugin:refresh',
+        ]);
+        $application = new Application($this->kernel);
+        $application->setAutoExit(false);
+        $application->run($pluginRefresh, new NullOutput());
+    }
+
+    private function hasSecurityAdvisories(): bool
+    {
+        $cacheKey = \sprintf('security-advisories-%s', $this->shopwareVersion);
+        $cacheItem = $this->objectCache->getItem($cacheKey);
+
+        if ($cacheItem->isHit()) {
```

Comment:
```
$value = $cache->get('my_cache_key', function (ItemInterface $item): string {
    $item->expiresAfter(3600);

    // ... do some HTTP request or heavy computations
    $computedValue = 'foobar';

    return $computedValue;
});
```

looks better to read

https://symfony.com/doc/current/components/cache.html#cache-contracts

Reply:
done :-)